### PR TITLE
added option to assign workload on rank

### PR DESCRIPTION
**Pull request by guj**
Added an command line option to assign workload per rank. This is relevant to ADIOS2 to determine how data is distributed in subfiles.

### CODE REVIEW

File: examples/8a_benchmark_write_parallel.cpp [hunk 1]
```diff
@@ -460,15 +465,19 @@ int parseArgs( int argc, char *argv[], TestInput& input )
...
         input.m_Steps = atoi( argv[4] );
 
-
     if (argc >= 6)
       input.m_Dim = atoi( argv[5] );
 
-    //if ( (dataDim > 3) || (dataDim < 0) )  {
-    //return -1;
-    //}
+    if (argc >= 7) {
```
Comment (ax3l) [inline examples/8a_benchmark_write_parallel.cpp:482]: Please don't forget to document the new option in `docs/source/usage/benchmarks.rst` so people know how to use it :)

File: examples/8a_benchmark_write_parallel.cpp [hunk 2]
```diff
@@ -310,9 +310,16 @@ class TestInput
 
   int m_MPISize = 1; //!< MPI size
   int m_MPIRank = 0; //!< MPI rank
+
   unsigned long m_XBulk = 64ul; //!< min num of elements at X dimension
   unsigned long m_YBulk = 32ul;  //!< min num of elements at Y dimension
   unsigned long m_ZBulk = 32ul;
+
+  /**  relative expansion of min grid(m_XBulk, m_YBulk, m_ZBulk)
+   *   to form a max grid. by fault max:min=1
```
Comment (ax3l) [inline examples/8a_benchmark_write_parallel.cpp]: I don't fully understand this part yet, but maybe the additional doc will solve that.

File: examples/8a_benchmark_write_parallel.cpp (hunk 1 above)
Comment (guj) [reply examples/8a_benchmark_write_parallel.cpp:482]: This option is making sense for bp users. The benchmark right now has fast track options for H5 users.  Will need to add another section later.

File: examples/8a_benchmark_write_parallel.cpp (hunk 2 above)
Comment (guj) [reply examples/8a_benchmark_write_parallel.cpp]: Added more comments there. Mainly this option is to direct range of workload per rank. Maybe the name is not fitting really well.

**Comment by ax3l** [comment]
@guj I rebased your branch because it was outdated and thus hard to review.

Please make sure to always update to the latest `dev` branch when starting new branches :)

**Review by ax3l — APPROVED** [review]
Thank you, Junmin! :christmas_tree:

**Pull request merged**